### ISSUE TITLE
Cannot rewrite date in blosc2 with fletcher32

### ISSUE THREAD

**Issue by berlin2123** [PyTables/PyTables#1014]
Cannot rewrite date when using blosc2 with fletcher32=true,  in pytables version: 3.8.0.
here it the test code:
```
[user@host tables_test]$ cat blosc.py
import numpy as N
import tables


hdf5_file_table = tables.open_file('data.hdf5', mode='w')

#filters = tables.Filters(complevel=5, complib='blosc', fletcher32=True)          # using this, works fine
#filters = tables.Filters(complevel=5, complib='blosc2')                          # using this, works fine
filters = tables.Filters(complevel=5, complib='blosc2', fletcher32=True)          # using this, err

NN=10

earrayDataA = hdf5_file_table.create_earray(
    hdf5_file_table.root,
    'dataA',
    tables.atom.ComplexAtom( itemsize=16, shape=(), dflt=0j ),   #
    shape=(0, NN, NN),
    filters=filters,
    expectedrows=10000000
    )

earrayDataA.append( N.zeros([1000, NN, NN], complex) )

hdf5_file_table.close()


hdf5_file_table = tables.open_file('data.hdf5', mode='a')

earrayDataA = hdf5_file_table.root.dataA

earrayDataA[:,:,0:NN-1] = N.ones([1000, NN, NN-1], complex)


hdf5_file_table.close()

```
and the err log :
```
Traceback (most recent call last):
  File "/home/user/tables_test/blosc.py", line 31, in <module>
    earrayDataA[:,:,0:NN-1] = N.ones([1000, NN, NN-1], complex)
  File "/home/user/.local/lib/python3.9/site-packages/tables/array.py", line 693, in __setitem__
    self._write_slice(startl, stopl, stepl, shape, nparr)
  File "/home/user/.local/lib/python3.9/site-packages/tables/array.py", line 775, in _write_slice
    self._g_write_slice(startl, stepl, countl, nparr)
  File "tables/hdf5extension.pyx", line 1807, in tables.hdf5extension.Array._g_write_slice
tables.exceptions.HDF5ExtError: HDF5 error back trace

  File "H5Dio.c", line 291, in H5Dwrite
    can't write data
  File "H5VLcallback.c", line 2113, in H5VL_dataset_write
    dataset write failed
  File "H5VLcallback.c", line 2080, in H5VL__dataset_write
    dataset write failed
  File "H5VLnative_dataset.c", line 207, in H5VL__native_dataset_write
    can't write data
  File "H5Dio.c", line 775, in H5D__write
    can't write data
  File "H5Dchunk.c", line 2703, in H5D__chunk_write
    unable to read raw data chunk
  File "H5Dchunk.c", line 3939, in H5D__chunk_lock
    data pipeline read failed
  File "H5Z.c", line 1390, in H5Z_pipeline
    filter returned failure during read
  File "hdf5-blosc2/src/blosc2_filter.c", line 244, in blosc2_filter
    Get chunk error

End of HDF5 error back trace

Internal error modifying the elements (H5ARRAYwrite_records returned errorcode -6)
Closing remaining open files:data.hdf5...done

```

**Comment by berlin2123** [comment]
Note that, there will no err when change the code to using blosc with fletcher32 or just using blosc2 without fletcher32.
that is, just change the comment to using the firtst or second line of the fellow part:
```
filters = tables.Filters(complevel=5, complib='blosc', fletcher32=True)             # using this, works fine
filters = tables.Filters(complevel=5, complib='blosc2')                                      # using this, works fine 
#filters = tables.Filters(complevel=5, complib='blosc2', fletcher32=True)         # using this, err 
```

**Comment by avalentino** [comment]
This should be now fixed in the PyTables 3.10
